collectors/monitoring_collector.go

```diff
@@ -84,11 +54,28 @@ type MonitoringCollector struct {
 	logger                          log.Logger
 }
 
-func NewMonitoringCollector(projectID string, monitoringService *monitoring.Service, filters map[string]bool, logger log.Logger) (*MonitoringCollector, error) {
-	if *monitoringMetricsTypePrefixes == "" {
-		return nil, errors.New("Flag `monitoring.metrics-type-prefixes` is required")
-	}
+type MonitoringCollectorOptions struct {
+	// MetricTypePrefixes are the Google Monitoring (ex-Stackdriver) prefixes metric type prefixes that the collector
+	// will be querying.
+	MetricTypePrefixes []string
+	// ExtraFilters is a list of extra filters to apply to each corresponding metric prefix query. If one or more are
+	// applicable to a given metric type prefix, they will be 'AND' concatenated.
+	ExtraFilters []MetricFilter
+	// RequestInterval is the time interval used in each request to get metrics. If there are many data points returned
+	// in during this interval, only the latest will be reported.
+	RequestInterval time.Duration
+	// RequestOffset is used to offset the requested interval into the past.
+	RequestOffset time.Duration
+	// IngestDelay decides if the ingestion delay specified in the metrics metadata is used when calculating the
+	// request time interval.
+	IngestDelay bool
+	// FillMissingLabels decides if empty labels should be filled with an empty string.
```

Comment:
```suggestion
	// FillMissingLabels decides if metric labels should be added with empty string to prevent failures due to label inconsistency on metrics.
```